src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.vb

```diff
@@ -114,6 +126,41 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return returnValue
         End Function
 
+        Private Shared Function IsFloatPointExpressionOfUnknownPrecision(rewrittenNode As BoundExpression) As Boolean
+            If rewrittenNode Is Nothing Then
+                Return False
+            End If
+
+            Dim Type = rewrittenNode.Type
+            If Type.SpecialType <> SpecialType.System_Double AndAlso Type.SpecialType <> SpecialType.System_Single Then
+                Return False
+            End If
+
+            Select Case rewrittenNode.Kind
```

Comment:
This method should return `false` if the expression has a constant value, as those should be computed and retained with correct precision.

Reply:
@gafter Thanks. I will fix the handling of constant value shortly.
I'd removed that case because I didn't manage to hit it, but I think I found a way now.